### PR TITLE
feat(cli): Refactoring the pull request creation to use git commands and support vector map update.

### PR DESCRIPTION
**Pull request by Wentao-Kuang**

### CODE REVIEW

File: packages/cli/src/cli/github/github.ts [hunk 1]
```diff
@@ -1,128 +1,73 @@
...
-  path: string;
-  mode: '100644';
-  type: 'blob';
-  sha: string;
-}
+import { execFileSync } from 'child_process';
 
 export class Github {
-  octokit: Api;
+  repo = 'basemaps-config';
```
Comment (blacha) [inline packages/cli/src/cli/github/github.ts:5]: we shouldnt hard code this, but if we are we should also include the org prefix. `linz/basemaps-config`

File: packages/cli/src/cli/github/make.cog.pr.ts [hunk 2]
```diff
@@ -10,54 +10,30 @@ export class MakeCogGithub extends Github {
...
   /**
    * Prepare and create pull request for the aerial tileset config
    */
-  async createTileSetPullRequest(
-    layer: ConfigLayer,
-    jira: string | undefined,
-    category: Category,
-  ): Promise<number | undefined> {
+  async updateRasterTileSet(config: string, layer: ConfigLayer, category: Category): Promise<void> {
+    const branch = `feat/config-raster-${this.imagery}`;
```
Comment (blacha) [inline packages/cli/src/cli/github/make.cog.pr.ts:17]: I thought we needed a common prefix to automatically create pull-requests? or is the common prefix `feat/config-`?

Comment (Wentao-Kuang) [reply packages/cli/src/cli/github/make.cog.pr.ts:17]: I decide to use `feat/config-*`

File: packages/cli/src/cli/github/make.cog.pr.ts [hunk 3]
```diff
@@ -127,4 +103,44 @@ export class MakeCogGithub extends Github {
 
     return tileSet;
   }
+
+  /**
+   * Prepare and create pull request for the aerial tileset config
+   */
+  async updateVectorTileSet(config: string, layer: ConfigLayer): Promise<void> {
```
Comment (blacha) [inline packages/cli/src/cli/github/make.cog.pr.ts:110]: `config` is pretty generic, what config is this related to?

this needs either a doc or a better name or both?

Comment (Wentao-Kuang) [reply packages/cli/src/cli/github/make.cog.pr.ts:110]: I think I will change this to `filename`

File: packages/cli/src/cli/github/github.ts [hunk 4]
```diff
@@ -1,128 +1,73 @@
...
-  path: string;
-  mode: '100644';
-  type: 'blob';
-  sha: string;
-}
+import { execFileSync } from 'child_process';
 
 export class Github {
-  octokit: Api;
+  repo = 'linz/basemaps-config';
```
Comment (blacha) [inline packages/cli/src/cli/github/github.ts:5]: could we do `--repo linz/basemaps-config` and default to `linz/basemaps-config`

File: packages/cli/src/cli/github/github.ts [hunk 5]
```diff
@@ -1,128 +1,73 @@
...
-  async createBlobs(content: string, path: string): Promise<Blob> {
-    // Create the blobs with the files content
-    this.logger.info({ path }, 'GitHub: Create blob');
-    const blobRes = await this.octokit.rest.git.createBlob({ owner, repo, content, encoding: 'utf-8' });
-    if (!this.isOk(blobRes.status)) throw new Error(`Failed to create data blob.`);
-
-    const blobSha = blobRes.data.sha;
-    return { path, mode: '100644', type: 'blob', sha: blobSha };
+  getBranch(branch: string): string {
+    // const basePath = path.dirname(repo);
```
Comment (blacha) [inline packages/cli/src/cli/github/github.ts:28]: commented out code

File: packages/cli/src/cli/github/make.cog.pr.ts (hunk 2 above)
Comment (blacha) [reply packages/cli/src/cli/github/make.cog.pr.ts:17]: could we maybe use `feat/bot-config` so we separate out bot commits vs human commits

**Review by blacha — APPROVED** [review]
